inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -285,3 +300,51 @@ def refextract(obj, eng):
     elif len(text_references) > len(pdf_references):
         obj.log.info('Extracted %d references from text.', len(text_references))
         obj.data['references'] = text_references
+
+
+def _normalize(s):
+    if not s:
+        return
+
+    result = RE_ALPHANUMERIC.sub(' ', s)
+    result = ' '.join(result.split())
+    result = result.upper()
+
+    if not result:
+        return
+    return result
```

Comment:
`return` will return `None`, so you don't need to differentiate between the two cases :)

Reply:
But `''` (empty string) is falsy without being `None`, so this change would change behavior.